### ISSUE TITLE
Installation error on Google Colab and local Ubuntu 20.04

### ISSUE THREAD

**Issue by souravjamwal77** [pycaret/pycaret#1573]
**Couldn't install the PyCaret on Google Colab and Local Ubuntu 20.04**
This is related to unmatched dependencies, as pyLDAvis 3.3.1 (latest) requires numpy>=1.20 and yellowbrick requires numpy<=1.20, which is creating a conflict for installing NumPy. I installed PyCaret on my system by downgrading pyLDAvis to 3.2.2 

**To Reproduce**
- just try to install PyCaret in a new virtualenv or on Google Colab
I tried on python 3.7 and 3.8, both.
```python
pip install pycaret
```
I ran this within virtual environment on my local but failed on NumPy version resolution.
and on Google Colab it failed on multiple packages, below are those packages

Google Colab Ouput:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires requests~=2.23.0, but you have requests 2.26.0 which is incompatible.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.
albumentations 0.1.12 requires imgaug<0.2.7,>=0.2.5, but you have imgaug 0.2.9 which is incompatible.

```

**Expected behavior**
PyCaret's dependencies are not locked as per SemVer and it is trying to download the latest version of each package whose required version is not mentioned which is causing the problem as different packages have similar dependencies with different versions.

**Additional context**

We should pin the dependencies in requirements.txt with SemVer notation like ~= which will let the bug updates to be updated but will ignore the minor and major versions.

**Versions**
<details>

pycaret version: 2.3.3
python versions: 3.7, 3.8
</details>

**Proposed Solution**
We should pin the dependencies in the _requirements.txt_ file with **SemVer notation like ~=**, so that only bug releases could update and no major and minor version could update

**Comment by moezali1** [comment]
@souravjamwal77 I just tried installing on Google Colab. I got warnings but the installation works fine:

![image](https://user-images.githubusercontent.com/54699234/133264398-6a2d4cf2-78e4-4510-a344-9015e3c160d2.png)

**Comment by souravjamwal77** [comment]
@moezali1, yes it's installed on Google Colab, but running on unmet dependencies still concerns me.
Also, for local testing, I think you're using windows, I tried on Ubuntu with pyenv for python versions(3.7 and 3.8) and for virtualenv I used venv and pipenv both but failed on NumPy on both of these.
You've pyLDAvis=3.2.2 installed on windows but for Linux it automatically installs 3.3.1 which has dependency NumPy>=1.20.0 and for yellowbrick it's <1.20 which is conflicting with each other.

I think we should pin the versions using SemVer notation like ~=3.2.0 it will update the bug update '0' but keep the major and minor version 3.2 locked.
Shall I make a PR for this? or you can look into this once? Because, if any of the dependencies released major version and PyCaret's dependencies were added with no version info then there may be breaking changes like in this case wherein pyLDAvis 3.3.0 had dependency NumPy with version >=1.20.0 and it broke my installation.

**Comment by moezali1** [comment]
@Yard1 @mavs @ngupta23 What are your thoughts on this?

**Comment by Yard1** [comment]
@souravjamwal77 A PR would be appreciated, thanks!

**Comment by souravjamwal77** [comment]
@Yard1 I'm working in it and will submit PR by day end. Also, I only have linux and Google colab, so, you or someone else have to test that on windows and Mac.

**Comment by xandrade** [comment]
I can see that issue is till there. The only workaround is using: ```pip install --use-deprecated=legacy-resolver pycaret``` 

But this doesn't resolve the issue as most of the other packages are downgraded (including pandas, scikit-learn, etc.)

**Comment by souravjamwal77** [comment]
Hi @xandrade 
Can you try with poetry or conda once?
Yes, pip and pipenv produce error because ther are conflicting dependencies but poetry and conda will install seperate package versions for you.

**Comment by xandrade** [comment]
> Are you seeing these issues on Google Colab or Ubuntu?

I'm using Google Colab

**Comment by ngupta23** [comment]
Can you share your colab notebook that is giving issues?

I am able to install it and run through examples without any issues - see this [colab notebook](https://nbviewer.org/gist/ngupta23/3078c5aac7842cbdc3e697a6307a6da6)
